### PR TITLE
[Fix #7029] Make `Style/Attr` not flag offense for custom `attr` method

### LINKED ISSUE #7029: Style/Attr is broken when an "attr" method is defined
if I have a method called `attr`, e.g. `attr(*args, &block)` or something like that then it breaks this cop.

--------

## Expected behavior

maybe it could only inspect classes/modules body?

## Actual behavior

inspects methods body

## Steps to reproduce the problem

source:

```ruby
class A
  def attr(*args)
    p args
  end

  def a
    attr(1)
    attr(:a, 1)
    attr(:b) { 2 }
    attr(:c) do |a|
      2
    end
  end
end
```

auto-corrected:

```ruby
class A
  def attr(*args)
    p args
  end

  def a
    attr_reader(1)
    attr_reader(:a, 1)
    attr_reader(:b) { 2 }
    attr_reader(:c) do |_a|
      2
    end
  end
end
```

(not sure how but I also had some `attr_accessor` and `attr_writer` inserted in my code, it somehow generated invalid ruby even like `attr_writer(:something`)

## RuboCop version

```
$ rubocop -V
0.68.1 (using Parser 2.6.3.0, running on ruby 2.6.3 x86_64-darwin18)
```

### PR DESCRIPTION
**Pull request by tejasbubane**
Closes #7029 

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/style/attr.rb [hunk 1]
```diff
@@ -21,6 +21,7 @@ class Attr < Cop
 
         def on_send(node)
           return unless node.command?(:attr) && node.arguments?
+          return if node.parent && !node.parent.class_type?
```
Comment (bbatsov) [inline lib/rubocop/cop/style/attr.rb]: So, the idea here is to check you're not calling the method in the class/module body, right? Might be a good idea to add a note explaining it.

**Comment by bbatsov** [comment]
I'm OK with the proposed fix, although I really think it's generally a bad idea to define methods with such names (at they might confuse people). I guess most of the time it happens as people simply don't know they exist.

**Comment by bbatsov** [comment]
Ping :-)

File: lib/rubocop/cop/style/attr.rb (hunk 1 above)
Comment (Drenmi) [reply lib/rubocop/cop/style/attr.rb]: I believe this will create some false negatives, for example in `#class_eval` blocks?

Comment (tejasbubane) [reply lib/rubocop/cop/style/attr.rb]: Thanks I missed that! I just made a change to consider `class_eval` & `module_eval`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,10 @@
 * [#7077](https://github.com/rubocop-hq/rubocop/issues/7077): **(Breaking)** Further standardisation of cop names. ([@scottmatthewman][])
 * [#7469](https://github.com/rubocop-hq/rubocop/pull/7469): **(Breaking)** Replace usages of the terms `Whitelist` and `Blacklist` with better alternatives. ([@koic][])
 
+### Bug fixes
+
+* [#7029](https://github.com/rubocop-hq/rubocop/issues/7029): Make `Style/Attr` not flag offense for custom `attr` method. ([@tejasbubane][])
```
Comment (koic) [inline CHANGELOG.md]: Please list changelog section in the order `### Bug fixes`, `### Changes`.

Comment (tejasbubane) [reply CHANGELOG.md]: done

**Comment by tejasbubane** [comment]
The JRuby failures look unrelated to this change.

**Comment by bbatsov** [comment]
@tejasbubane For some reason I can't rebase this from the GitHub UI. Can you rebase it again and ping me to merge it?

**Comment by tejasbubane** [comment]
@bbatsov Rebased.

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**